File: cpg-solidity/src/main/java/de/fraunhofer/aisec/cpg/frontends/solidity/nodes/Revert.java
package de.fraunhofer.aisec.cpg.frontends.solidity.nodes;

import de.fraunhofer.aisec.cpg.graph.SubGraph;
import de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression;
import org.jetbrains.annotations.Nullable;
import org.neo4j.ogm.annotation.Relationship;

public class Revert extends Expression {
    @Relationship(value = "MESSAGE", direction = "OUTGOING")
    @SubGraph("AST")
    @Nullable
    private Expression message;

    @Nullable
    public final Expression getMessage() {
        return this.message;
    }

    public final void setMessage(@Nullable Expression var1) {
        this.message = var1;
    }
}

File: cpg-solidity/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/solidity/nodes/Revert.kt
package de.fraunhofer.aisec.cpg.frontends.solidity.nodes

import de.fraunhofer.aisec.cpg.graph.AST
import de.fraunhofer.aisec.cpg.graph.statements.expressions.Expression
import org.neo4j.ogm.annotation.Relationship

class Revert : Expression() {
    @field:Relationship(value = "MESSAGE", direction = Relationship.Direction.OUTGOING)
    @AST
    var message: Expression? = null
        set(var1) {
            field = var1
            message!!.addNextDFG(this)
        }
}